### DISCUSSION TITLE
How to restrict patch to specific PHP version?

### DISCUSSION THREAD

**Discussion by tenzap** [cweagans/composer-patches#598] [Q&A] [answered]
Question is in the title. 

For example I would like to apply a patch only is the user is running php 8.4 or above.

**Comment by cweagans** [comment, marked as answer]
There is not a way to do that. You can write a patch resolver plugin to make that work, but a more straightforward option might be to change the patch so that whatever depends on a higher php version is conditionally invoked at runtime (`if (version_compare(...)) { ... }`).

**Reply by tenzap** [reply]
Thanks. Unfortunately it is not always possible to use this conditional especially when it is about patching breaking PHP language changes betweens versions. Fortunately for this time I could find workarounds.
